### PR TITLE
Fixes #19820, #19822, #19824 - Fix some performance cops

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

**Review by akofink — APPROVED** [review]
LGTM - APT

**Comment by mmoll** [comment]
[test upgrade]

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -392,27 +392,12 @@ Naming/VariableNumber:
 Performance/Caller:
   Enabled: false
 
-# Offense count: 9
-# Cop supports --auto-correct.
-Performance/Casecmp:
-  Enabled: false
```
Comment (ares) [inline .rubocop_todo.yml:382]: I don't like this one too much, is there a chance to keep it disabled? How big performance boost it brings?

Comment (tbrisker) [reply .rubocop_todo.yml:382]: It depends on the length of the string, some quick benchmarks indicate about 1.2x faster for 4 letter string and about 2.2x faster for comparing a 50 letter string to a 4 letter string. Considering afaict none of these occurrences are in a very critical path that gets called very rapidly I'd say we can ignore this cop. I'll revert these changes.

Comment (tbrisker) [reply .rubocop_todo.yml:382]: fixed.

**Comment by mmoll** [comment]
test failure unrelated

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
merged, thanks @tbrisker!

**Comment by lzap** [comment]
Nice thanks :-)

**Pull request merged**